### ISSUE TITLE
 Serialization and deserialization of 'System.Action' instances are not supported. error

### ISSUE THREAD

**Issue by LasseBuchAspIT** [Kaliumhexacyanoferrat/GenHTTP#459]
Hi.

I'm Getting a  "Serialization and deserialization of 'System.Action' instances are not supported." error when attempting to create an async  HTTPget returning Task<Obj>

![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/97156204/93eac8b1-50b4-49f0-a26f-24e7620fb45e)

is there a solution for this?

**Comment by LasseBuchAspIT** [comment]
Yes ofc 

Unfortunately i am done with work for the day, but i will send the stack trace and some more info tomorrow :)

**Comment by LasseBuchAspIT** [comment]
Heres the full stacktrace

![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/97156204/1e094b63-c16e-43bf-841d-43c3ccbf89f8)
![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/97156204/33146e98-f0ce-4caf-9685-a884f59fc2da)

**Comment by Kaliumhexacyanoferrat** [comment]
It seems that you have an `Action` somewhere in your `QuickBtn` structure which cannot get serialized by the JSON formatter. You can mark this property with `[JsonIgnore]`, so it will not get serialized. See the [documentation](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/ignore-properties) for details.

**Comment by LasseBuchAspIT** [comment]
Hi again

im pretty confused as to where this action could be as my quickBtn class is very simple

![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/97156204/221f2e15-9980-4bff-a451-2c73bb5ac6c4)

**Comment by Kaliumhexacyanoferrat** [comment]
Hm, I updated my sample:

```csharp
var auth = ApiKeyAuthentication.Create()
                               .Keys("ABC-123", "BCD-234");

var app = Layout.Create()
                .AddService<QuickService>("quick")
                .Authentication(auth);

Host.Create()
    .Handler(app)
    .Defaults()
    .Development()
    .Console()
    .Run();

public class QuickService
{

    [ResourceMethod("GetQuickBtns")]
    public async Task<List<QuickBtn>> GetQuickBtn(IRequest req)
    {
        List<QuickBtn> returnList = new();

        try
        {
            returnList = await GetResult(req.GetUser<ApiKeyUser>()!);
        }
        catch
        {
            return new();
        }

        return returnList;
    }

    private Task<List<QuickBtn>> GetResult(ApiKeyUser user)
    {
        return Task.FromResult(new List<QuickBtn>()
        {
            new QuickBtn() { Id = 1, UserId = 4711, Name = "Me", Script = "alert('Hello World');", User = user }
        });
    }


}

public partial class QuickBtn
{

    public int Id { get; set; }

    public int UserId { get; set; }

    public string? Name { get; set; }        

    public string? Script { get; set; }

    [JsonIgnore]
    public virtual ApiKeyUser? User { get; set; }

}
```

But still fail to reproduce this:

![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/4992119/caa02d6c-d65b-49c8-8d71-ead3ec05af37)

Does removing the `User` property from the response change anything? I see that your class is defined `partial`, are there any additions to this class somewhere else?

**Comment by LasseBuchAspIT** [comment]
Not that i know of, but the class was generated by microsoft entity framework, so there might be some hidden additions that i am unaware of

maybe try reverse engineering a class with efcore to replicate?

**Comment by Kaliumhexacyanoferrat** [comment]
Or just view an instance in the debugger, this will show you all the members. But yes, I guess EF is the issue here. I think one of the options in the context menu which opens when you click on the class will show you where the other partial class is located.

**Comment by LasseBuchAspIT** [comment]
After messing around a bit more ive found that the problem persists even when using a very simple non partial Test class 

![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/97156204/117ae1d9-20fd-4895-a2fe-4caf78656ff9)
![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/97156204/2b7acafc-5535-4cb8-a69d-5a4e8c22471e)

**Comment by LasseBuchAspIT** [comment]
it appears yhe bug dissapears when removing await from function, even if function is still declared as async

**Comment by Kaliumhexacyanoferrat** [comment]
Hm, which version of .NET do you use and how do you run this application? Plain console or something different like WPF?

**Comment by LasseBuchAspIT** [comment]
both .NET and Genhttp framework are running version 7.0
ill try updating to 8.0 and see if that fixes anything

**Comment by LasseBuchAspIT** [comment]
Updating did not fix.
This doesnt seem to be a GenHttp issue, thanks for the help regardless

**Comment by LasseBuchAspIT** [comment]
this appears to be a GenHttp problem after all.
below are two examples, one in AspNet and one in GenHttp

the code is essentially the same, except for a small settingsReader class, being used in the AspNET example
The AspNET version runs without issue, whereas the GenHttp version crashes...

GenHttp: 
![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/97156204/9f7c232a-e739-47ff-9afe-c235053a7b35)
![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/97156204/293e5906-f93b-45cb-b443-362edbe47395)

AspNET:
![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/97156204/16a871dd-270c-4ef0-b336-5ab3276487a4)
![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/97156204/74fbc31c-a659-4372-abf1-1f8f69e0ebc3)

**Comment by LasseBuchAspIT** [comment]
Updating to 8.0 did however get me a new error: 
![image](https://github.com/Kaliumhexacyanoferrat/GenHTTP/assets/97156204/9dd5a3f8-6c33-453b-9604-4e9a1b799ea5)

**Comment by Kaliumhexacyanoferrat** [comment]
Still fail to reproduce this. Can you ZIP and attach your `GenHttpAgain` sample here? Do not need connection strings or anything sensitive, just the minimal code to reproduce this issue.

**Comment by LasseBuchAspIT** [comment]
[GenHttpAgain.zip](https://github.com/Kaliumhexacyanoferrat/GenHTTP/files/14404106/GenHttpAgain.zip)
will this do?

**Comment by LasseBuchAspIT** [comment]
this example is running on a local Test server so i can send you a copy of the DB to if needed

**Comment by Kaliumhexacyanoferrat** [comment]
> [GenHttpAgain.zip](https://github.com/Kaliumhexacyanoferrat/GenHTTP/files/14404106/GenHttpAgain.zip) will this do?

Sorry, meant the source code and Visual Studio project. You can omit any code not related to the issue or sensitive in any way.

**Comment by LasseBuchAspIT** [comment]
https://github.com/LasseBuchAspIT/GenHttpAgain
you can find the source code here

ill send the project in a moment

**Comment by Kaliumhexacyanoferrat** [comment]
Found and fixed the problem in version 8.1.1 and above. Problem was caused by the return types used by EF for asynchronous operations. The release should now be available via nuget.

Thank you very much for reporting the problem and for your patience when reproducing the issue!

**Comment by LasseBuchAspIT** [comment]
thank you very much for the help 👍 was a pleasure
